lock-api/src/main/java/com/palantir/lock/client/timestampleases/TimestampLeaseAcquirerImpl.java

```diff
@@ -62,28 +64,38 @@ public final class TimestampLeaseAcquirerImpl implements TimestampLeaseAcquirer
     private final NamespacedTimestampLeaseService delegate;
     private final Unlocker unlocker;
     private final Supplier<UUID> uuidSupplier;
+    private final Counter notEnoughFreshTimestampsCounter;
 
     private final Retryer<Optional<TimestampLeaseResponses>> retryer =
             RetryerBuilder.<Optional<TimestampLeaseResponses>>newBuilder()
                     .retryIfResult(Optional::isEmpty)
                     .withStopStrategy(StopStrategies.stopAfterAttempt(3))
                     .build();
 
-    @VisibleForTesting
-    TimestampLeaseAcquirerImpl(
-            NamespacedTimestampLeaseService delegate, Unlocker unlocker, Supplier<UUID> uuidSupplier) {
+    private TimestampLeaseAcquirerImpl(
+            NamespacedTimestampLeaseService delegate,
+            Unlocker unlocker,
+            Supplier<UUID> uuidSupplier,
+            Counter notEnoughFreshTimestampsCounter) {
         this.delegate = delegate;
         this.unlocker = unlocker;
         this.uuidSupplier = uuidSupplier;
+        this.notEnoughFreshTimestampsCounter = notEnoughFreshTimestampsCounter;
     }
 
-    private TimestampLeaseAcquirerImpl(NamespacedTimestampLeaseService delegate, Unlocker unlocker) {
-        this(delegate, unlocker, UniqueIds::pseudoRandomUuidV4);
+    @VisibleForTesting
+    TimestampLeaseAcquirerImpl(
+            NamespacedTimestampLeaseService delegate, Unlocker unlocker, Supplier<UUID> uuidSupplier) {
+        this(delegate, unlocker, uuidSupplier, new Counter());
     }
 
     public static TimestampLeaseAcquirer create(
-            NamespacedTimestampLeaseService timestampLeaseService, LockTokenUnlocker unlocker) {
-        return new TimestampLeaseAcquirerImpl(timestampLeaseService, identifier -> unlock(unlocker, identifier));
+            NamespacedTimestampLeaseService delegate, LockTokenUnlocker unlocker, TimestampLeaseMetrics metrics) {
+        return new TimestampLeaseAcquirerImpl(
+                delegate,
+                identifier -> unlock(unlocker, identifier),
+                UniqueIds::pseudoRandomUuidV4,
```

Comment:
Generally in the rest of the code we use `UUID::randomUUID`. This is plausibly better, but was there a reason for using it here specifically?